src/routers/user.ts

```diff
@@ -0,0 +1,80 @@
+import express from 'express';
+import { UserService } from '../services/userService';
+import {
+  userBodyValidatorOnCreate,
+  userBodyValidatorOnUpdate,
+  userParamsIdValidator,
+  userQuerySubstringLimitValidator,
+} from '../validation/validators';
+import { User } from '../types';
+
+const user = express.Router();
+const userService = new UserService();
+
+user.post('/user', userBodyValidatorOnCreate, async (req, res) => {
+  const user: User = { ...req.body };
+  const result = await userService.createUser(user);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    res.status(200).json({ createdUser: result });
+  }
```

Comment:
This should be done in error handler (module 5). But errors aren't returned, they are thrown. And thrown errors should be handled in one error handler. See [https://github.com/goldbergyoni/nodebestpractices#2-error-handling-practices](https://github.com/goldbergyoni/nodebestpractices#2-error-handling-practices)

Reply:
I'll fix that in module 5